cli/pkg/cmdutil/githelper.go

```diff
@@ -0,0 +1,200 @@
+package cmdutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/rilldata/rill/admin/client"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/semaphore"
+)
+
+var (
+	gitignoreHasDotenvRegexp       = regexp.MustCompile(`(?m)^\.env$`)
+	gitignoreHasDotRillCloudRegexp = regexp.MustCompile(`(?m)^\s*\.rillcloud/`)
+)
+
+// GitHelper manages git operations for a project
+// It also caches the git credentials for the project
+type GitHelper struct {
+	c         *client.Client
```

Comment:
I think it should hold a reference to `cmdutil.Helper` instead here, and then use it to get the client when it needs it.

It isn't safe to keep the admin client cached/open outside of `cmdutil.Helper`, see the logic here: https://github.com/rilldata/rill/blob/7e9ec7c223146c0f13211de822467183d7bb4ef2/cli/pkg/cmdutil/helper.go#L142